File: api/Models/Message.cs
using System;
using api.Enums;

namespace api.Models
{
    public class Message
    {
        public string UserName { get; set; }
        public string Content { get; set; }
        public EMessageType Type { get; set; }
        public DateTime CreatedAt { get; set; }

        public Message(string userName, string content, EMessageType type)
        {
            UserName = userName;
            Content = content;
            Type = type;
            CreatedAt = DateTime.Now;
        }
    }
}

File: web/src/app/models/message.ts
export enum EMessageType {
  Log = 1,
  Text = 2,
  Photo = 3,
  Video = 4,
  Url = 5
}

export class Message {
  userName: string;
  content: string;
  type: EMessageType;
  createdAt: Date;

  constructor(content: string, type: EMessageType) {
    this.content = content;
    this.type = type;
  }
}
